src/Main.cpp

```diff
@@ -65,6 +64,33 @@ class BittyBuzzTask : public AbstractTask<6 * configMINIMAL_STACK_SIZE> {
     }
 };
 
+class UARTReadTask : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
+  public:
+    UARTReadTask(const char* taskName, UBaseType_t priority) :
+        AbstractTask(taskName, priority), m_logger(LoggerContainer::getLogger()) {}
+
+    ~UARTReadTask() override = default;
+
+  private:
+    ILogger& m_logger;
+    void task() override {
+
+        uint8_t buffer[5];
+        buffer[sizeof(buffer) - 1] = 0;
```

Comment:
Shouldn't you null terminate after receiving? If it's for a string null termination, this way would be more verbose
`buffer[sizeof(buffer)-1] = '\0'`

Reply:
Changed for '\0', but honestly most of the code in the main will disappear pretty soon.